### PR TITLE
Update readme docs to replace deprecated token generate functions with new ones which handle opt-out

### PR DESCRIPTION
**Pull request by jon8787**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,24 +92,24 @@ If you're using server-only integration (see [Publisher Integration Guide, Serve
...
+    `tokenGenerateResponse.getIdentityJsonString()` //Note: this method will return null if the user has opted out, so be sure to handle that case.
 
 #### Server-Only Integration
 
 If you're using server-only integration (see [Publisher Integration Guide, Server-Only](https://unifiedid.com/docs/guides/custom-publisher-integration)):
 
-1. Store this identity as a JSON string in the user's session, using: `identity.getJsonString()`.
-2. To use the user's UID2 token, use `identity.getAdvertisingToken()`.
+1. Store this identity as a JSON string in the user's session, using: `tokenGenerateResponse.getIdentityJsonString()`. This method will return null if the user has opted out, so be sure to handle that case.
+2. To use the user's UID2 token, use `IdentityTokens identity = tokenGenerateResponse.getIdentity()` followed by `identity.getAdvertisingToken()`.
```
Comment (atarassov-ttd) [inline README.md:112]: Should this mention possibility of null too?

Comment (genwhittTTD) [reply README.md:112]: Best to use present tense, simpler and conveys the same info: "this method returns null if the user has opted out..." But your call. I am approving the review but that's my suggestion.

File: README.md [hunk 2]
```diff
@@ -44,25 +44,33 @@ If you're using the SDK's HTTP implementation, follow these steps.
...
-2. To use the user's UID2 token, use the `identity.getAdvertisingToken()` function.
-3. When the user accesses another page, or on a timer, determine whether a refresh is needed:
+1. Store this identity as a JSON string in the user's session, using the `tokenGenerateResponse.getIdentityJsonString()` function. This method returns `null` if the user has opted out, so be sure to handle that case.
+2. To retrieve the user's UID2 token, use:
+
+   ```
+   IdentityTokens identity = tokenGenerateResponse.getIdentity();
+   if (identity != null) { String advertisingToken = identity.getAdvertisingToken(); }
+   ```
+4. When the user accesses another page, or on a timer, determine whether a refresh is needed:
```
Comment (sunnywu) [inline README.md:73]: should this be "3." instead of "4."?

File: README.md [hunk 3]
```diff
@@ -72,44 +80,53 @@ If you're using server-only integration (see [Publisher Integration Guide, Serve
    3. Determine if a refresh is needed:
 
       `if (identity.isDueForRefresh()) {..}`
-4. If needed, refresh the token and associated values:
+5. If needed, refresh the token and associated values:
```
Comment (sunnywu) [inline README.md:83]: 4. ?

File: README.md [hunk 4]
```diff
@@ -72,44 +80,53 @@ If you're using server-only integration (see [Publisher Integration Guide, Serve
...
  
    `TokenRefreshResponse tokenRefreshResponse = publisherUid2Client.refreshToken(identity);`
  
-5. Store `tokenRefreshResponse.getIdentityJsonString()` in the user's session. If the user has opted out, this method returns null, indicating that the user's identity should be removed from the session. To confirm optout, you can use the `tokenRefreshResponse.isOptout()` function.
+6. Store `tokenRefreshResponse.getIdentityJsonString()` in the user's session. If the user has opted out, this method returns `null`, indicating that the user's identity should be removed from the session. To confirm optout, you can use the `tokenRefreshResponse.isOptout()` function.
```
Comment (sunnywu) [inline README.md:87]: 5. instead of 6.?

File: README.md [hunk 5]
```diff
@@ -44,25 +44,33 @@ If you're using the SDK's HTTP implementation, follow these steps.
  
    `private final PublisherUid2Client publisherUid2Client = new PublisherUid2Client(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);`
 
-2. When the user has authenticated, and has authorized the creation of a UID2, call a function that takes the user's email address or phone number as input and generates an `IdentityTokens` object. The following example uses an email address:
- 
-   `IdentityTokens identity = publisherUid2Client.generateToken(TokenGenerateInput.fromEmail(emailAddress));`
+2. Call a function that takes the user's email address or phone number as input and generates a `TokenGenerateResponse` object. The following example uses an email address:
```
Comment (sunnywu) [inline README.md:47]: should this mention hashed email/phone number as well?

Comment (jon8787) [reply README.md:47]: I think that's a niche use-case. This readme doesn't mention every function, just the main ones

File: README.md (hunk 2 above)
Comment (jon8787) [reply README.md:73]: Ideally, but not urgent. We could also put all 1s there. It's fine in the rendered view: https://github.com/IABTechLab/uid2-client-java/pull/32/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

Comment (sunnywu) [reply README.md:73]: interesting it's fixed automagically!

**Review by sunnywu — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -44,25 +44,33 @@ If you're using the SDK's HTTP implementation, follow these steps.
...
  
+   `TokenGenerateResponse tokenGenerateResponse = publisherUid2Client.generateTokenResponse(TokenGenerateInput.fromEmail(emailAddress).doNotGenerateTokensForOptedOut());`
+
+   >IMPORTANT: Be sure to call this function only when you have obtained legal basis to convert the user’s [Directly Identifying Information (DII)](https://unifiedid.com/docs/ref-info/glossary-uid#gl-dii) to UID2 tokens for targeted advertising.
+   
+   >`doNotGenerateTokensForOptedOut()` applies `policy=1` in the [/token/generate](https://unifiedid.com/docs/endpoints/post-token-generate#token-generation-policy) call. Without this, `policy` is omitted to maintain backwards compatibility.
 #### Standard Integration
 
 If you're using standard integration (client and server) (see [Client-Side JavaScript SDK Integration Guide](https://unifiedid.com/docs/guides/publisher-client-side)), follow this step:
```
Comment (genwhittTTD) [inline README.md:56]: Publication name has changed to: UID2 SDK for JavaScript Integration Guide

If you're using standard integration (client and server) (see [UID2 SDK for JavaScript Integration Guide](https://unifiedid.com/docs/guides/publisher-client-side)), follow this step:

**Review by genwhittTTD — COMMENTED** [review]
One update which isn't part of your changes but just noticed it, publication name change. Hoping you could fix that since you're updating the file.

File: README.md (hunk 6 above)
Comment (jon8787) [reply README.md]: fixed, thanks

**Pull request merged**